book/data_viz/plotting_data.nim

```diff
@@ -7,18 +7,266 @@ nbUseNimibook
+will just list all kinds of `geoms` we wish to draw. The order we list them directly determines
+the order they are drawn in (the later ones are drawn on top of the former ones, which is important
+to remember for more busy plots).
+
+This is what all available `geom_*` procedures are for. They return `Geom` variant objects that mainly
+just store their kind and possibly some specific information required to draw them.
+
+The (currently) implemented geoms are as follows (with the required aesthetics listed):
+- `geom_point`: draw points for each `x`/`y`
+- `geom_line`: draw a line through all `x`/`y`
+- `geom_errorbar`: draw error bars from `xMin` to `xMax` or `yMin` to `yMax` at `x`/`y`
+- `geom_linerange`: draw lines from `xMin` to `xMax` or `yMin` to `yMax`
+- `geom_bar`: draw a *discrete* bar plot using the occurrences (default `stat = "count"`) of each
+   discrete value in `x` or the number of counts indicated in `y` (`stat = "identity"`)
+- `geom_histogram`: draw a *continuous* bar plot computing a histogram from continuous variable `x`
+   (default `stat = "bin"`) or draw continuous bars starting at `x` and the number of entries
+   indicated in `y` (`stat = "identity"`).
+- `geom_freqpoly`: same as `geom_histogram`, but connect bin centers by lines instead of drawing bars
+- `geom_tile`: draw discrete tiles at `x`/`y` (default position bottom left) with width `width` and height
+  `height` each. Tiles don't need to touch.
+- `geom_raster`: draw fully connected tiles at `x`/`y` of `width` and `height`. `width` and `height` must
+   be constant!
+- `geom_text`: draw text at `x`/`y` containing `text` (the `text` aesthetic)
+
+Here we stated mainly the *typical* (or default) use cases. All geoms take all arguments. That means
+you can also draw a histogram using points by applying the `stat = "bin"` argument. The difference
+is just in the defaults! Or in case of a `geom_histogram` call you can indicate that the `binPosition`
+should be `"center"` instead of the default `"left"` to have `x` indicate the bin centers.
+
+The possibilities are almost endless. You can combine any geom with (almost) any option and
+it *should just work* (few exceptions exist, e.g. `geom_raster` only draws fixed size tiles for performance).
+
+## Another look at *that* plot
+
+Let's look at the plot from above again. This time read the command with the gained knowledge and then
+see if the explanation in words makes more sense now.
 """
+nbCode:
+  import ggplotnim, random, sequtils
+  let df = seqsToDf({ "Energy" : cycle(linspace(0.0, 10.0, 25).toRawSeq, 2),
+                      "Counts" : concat(toSeq(0 ..< 25).mapIt(rand(10.0)),
+                                        toSeq(0 ..< 25).mapIt(rand(10).float)),
+                      "Type" : concat(newSeqWith(25, "background"),
+                                      newSeqWith(25, "candidates")) })
+  echo "Input data frame: "
```

Comment:
Missing the `df` or is it just a "header" of sorts for the head and tail on the lines below?

Reply:
That's just a header for the output of the two echoes below (on the preview it looks fine like this, imo.

Reply:
Okay 👍 I think it was mainly the space after the `:` that tricked me into believing that something should come after it 😅